src/transformers/data/data_tf_collator.py

```diff
@@ -0,0 +1,218 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, Optional, Tuple, Union
+
+import tensorflow as tf
+
+from ..tokenization_utils_base import BatchEncoding, PreTrainedTokenizerBase
+
+
+class TFDataCollatorForLanguageModeling:
+    """
+    Data collator used for language modeling. Encodes sequences for Masked Language Modeling as mentioned in the paper
+    'BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding'.
+
+    Labels are -100 for non-masked tokens and the value to predict the masked token.
+
+    Args:
+        tokenizer (:class:`~transformers.PreTrainedTokenizer` or :class:`~transformers.PreTrainedTokenizerFast`):
+            The tokenizer used for encoding the data.
+        batch_size (:obj:`int`):
+            The size of the batch.
+        padding_length (:obj:`int`, `optional`):
+            The length of the vector concatenated to the largest element.
+        mlm_probability (:obj:`float`, `optional`, defaults to 0.15):
+            The probability with which to (randomly) mask tokens in the input.
+        special_tokens_mask (:obj:`tf.Tensor`, `optional`):
+            If set, special tokens will have zero probability of being masked.
+    """
+
+    def __init__(
+        self,
+        tokenizer: PreTrainedTokenizerBase,
+        batch_size: int,
+        special_tokens_mask: tf.Tensor = None,
+        padding_length: int = None,
+    ):
+
+        self.tokenizer = tokenizer
+        self.batch_size = batch_size
+        self.padding_length = padding_length
+        self.special_tokens_mask = special_tokens_mask
+        self.mlm_probability = 0.15
+
+    def __post_init__(self):
+        if self.tokenizer.mask_token is None:
+            raise ValueError(
+                "This tokenizer does not have a mask token which is necessary for masked language modeling. "
+                "You should pass `mlm=False` to train on causal language modeling instead."
+            )
+
+    @tf.function
+    def pseudo_bernoulli(self, prob_matrix, labels):
+        return tf.cast(prob_matrix - tf.random.uniform(tf.shape(labels), 0, 1) >= 0, tf.bool)
+
+    @tf.function
+    def mask_special_tokens(self, labels, special_tokens):
+        # Finds all special tokens within labels
+        x = tf.map_fn(lambda b: tf.cast(tf.math.equal(labels, b), tf.int32), special_tokens)
+        return tf.math.greater(tf.reduce_sum(x, axis=0), 0)
+
+    @tf.function
+    def tf_pad_tokens(
```

Comment:
This looks like it's just reproducing `tokenizer.pad` - can I ask why you want to do it here instead of calling that function? So you can compile the whole thing into tf.function or a tf.data graph?